File: core/src/com/spacefist/components/Physics.java
package com.spacefist.components;

import com.spacefist.GameData;
import com.spacefist.components.abst.PhysicsComponent;
import com.spacefist.entities.Entity;
import org.jetbrains.annotations.NotNull;

/**
 * Updates The game objects position using its properties.
 */
public class Physics implements PhysicsComponent {
    @Override
    public void update(GameData gameData, @NotNull Entity obj) {
        obj.setX(obj.getX() + (int) obj.getVelocity().x);
        obj.setY(obj.getY() - (int) obj.getVelocity().y);
    }
}

File: core/src/com/spacefist/components/Physics.kt
package com.spacefist.components

import com.spacefist.GameData
import com.spacefist.components.abst.PhysicsComponent
import com.spacefist.entities.Entity

/**
 * Updates The game objects position using its properties.
 */
class Physics : PhysicsComponent {
    override fun update(gameData: GameData, obj: Entity) {
        obj.x = obj.x + obj.velocity.x.toInt()
        obj.y = obj.y - obj.velocity.y.toInt()
        obj.rotation = obj.rotation % 360
    }
}
